weave/ops_domain/artifact_type_ops.py

```diff
@@ -61,7 +62,14 @@ def op_artifact_type_name(artifactType: wdt.ArtifactType) -> str:
                 }}
             }}
         }}
-    }}"""
+    }}""",
+        gql_key_propagation_fn=lambda inputs, input_type: wdt.ArtifactVersionType.with_keys(
```

Comment:
I left a comment elsewhere, but just reading this again: it does feel like we would require this param for custom gql ops

Reply:
See above. In the case where it's none, we lose the type information and revert to unkeyed GQL types.